### PR TITLE
Allow user to pass own `tune.run` params in `fit`

### PR DESCRIPTION
**Pull request by Yard1**
Will fix https://github.com/ray-project/tune-sklearn/issues/211

### CODE REVIEW

File: tune_sklearn/tune_basesearch.py [hunk 1]
```diff
@@ -735,18 +739,37 @@ def _fill_config_hyperparam(self, config):
...
+        if tune_params:
+            user_overrides = {k for k in tune_params if k in run_args}
+            if user_overrides:
+                warnings.warn(
+                    "The following preset tune.run parameters will "
+                    f"be overriden by tune_params: {', '.join(user_overrides)}"
+                    ". This may cause unexpected issues! If you experience "
+                    "issues, please try removing those parameters from "
+                    "tune_params.")
+            warnings.warn("Using user supplied tune_params.")
```
Comment (richardliaw) [inline tune_sklearn/tune_basesearch.py:769]: hmm, if a user passes this in, they wouldn't expect a warning here right?

File: tune_sklearn/tune_basesearch.py [hunk 2]
```diff
@@ -735,18 +739,37 @@ def _fill_config_hyperparam(self, config):
...
+    def _override_run_args_with_tune_params(self, run_args, tune_params):
+        """Helper method to override tune.run run arguments with user
+        supplied dict"""
+
+        if tune_params:
+            user_overrides = {k for k in tune_params if k in run_args}
+            if user_overrides:
+                warnings.warn(
+                    "The following preset tune.run parameters will "
+                    f"be overriden by tune_params: {', '.join(user_overrides)}"
```
Comment (richardliaw) [inline tune_sklearn/tune_basesearch.py:765]: this is a great error message!

File: tune_sklearn/tune_search.py [hunk 3]
```diff
@@ -626,6 +626,8 @@ def _tune_run(self, config, resources_per_trial):
             resources_per_trial (dict): Resources to use per trial within Ray.
                 Accepted keys are `cpu`, `gpu` and custom resources, and values
                 are integers specifying the number of each resource to use.
+            tune_params (dict, optional): User defined parameters passed to
```
Comment (richardliaw) [inline tune_sklearn/tune_search.py:629]: I think to get this to properly render on readthedocs, you can just leave this as (dict)

Comment (richardliaw) [reply tune_sklearn/tune_search.py:629]: Can you add a comment here also about the overriding behavior?

File: tune_sklearn/tune_basesearch.py [hunk 4]
```diff
@@ -735,18 +739,37 @@ def _fill_config_hyperparam(self, config):
...
                 number of cpu's and gpu's to use to train the model.
+            tune_params (dict, optional): User defined parameters passed to
+                ``tune.run``.
 
         """
         raise NotImplementedError("Define in child class")
 
+    def _override_run_args_with_tune_params(self, run_args, tune_params):
+        """Helper method to override tune.run run arguments with user
+        supplied dict"""
```
Comment (richardliaw) [inline tune_sklearn/tune_basesearch.py:758]: nit (not important here but for future ref), the first line of the docstring is a summary line - and should not have a linebreak in the middle. this is important when building docs

**Comment by richardliaw** [comment]
Sorry I missed this! left some simple comments. Please assign me when you get around to addressing the issues!

File: tune_sklearn/tune_basesearch.py (hunk 1 above)
Comment (Yard1) [reply tune_sklearn/tune_basesearch.py:769]: This probably should be an logging info message instead of a warning, now that I think about it

**Comment by Yard1** [comment]
Thanks for letting me know about the docs!

**Review by richardliaw — APPROVED** [review]

**Pull request merged**